src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -192,7 +198,7 @@ public static String generateXpathQuery(DetailAST ast) {
             }
             xpathQueryBuilder.append(']');
         }
-        return xpathQueryBuilder.toString();
+        return NEWLINE_TO_TAG.matcher(xpathQueryBuilder.toString()).replaceAll("\\\\n");
```

Comment:
@romani This will convert multiple newlines to a single, hence the `+`. Is this intended as this will manipulate the text being matched against? `\n\n` will be displayed as `\n`.

Reply:
This is not indented, I am fixing it. Our aim to make Xpath similar to text attribute of `TEXT_BLOCK_CONTENT` of AST.

Reply:
Done